src/main/java/com/rtb/projectmanagementtool/private-comments/PrivateCommentListServlet.java

```diff
@@ -0,0 +1,102 @@
+/** Servlet responsible for privatecomments */
+package com.rtb.projectmanagementtool.privatecomment;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.*;
+import com.rtb.projectmanagementtool.auth.*;
+import com.rtb.projectmanagementtool.task.*;
+import com.rtb.projectmanagementtool.user.*;
+import java.io.IOException;
+import java.util.*;
+import java.util.ArrayList;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/private-comment")
+public class PrivateCommentListServlet extends HttpServlet {
+  DatastoreService datastore;
+  AuthOps auth;
+
+  public PrivateCommentListServlet() {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    auth = new AuthOps(datastore);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+
+    Long userID = Long.parseLong(request.getParameter("userID"));
+    Long currentUser = auth.whichUserIsLoggedIn(request, response);
+
+    // controllers
+    UserController userController = new UserController(datastore);
+    TaskController taskController = new TaskController(datastore);
+    PrivateCommentController privateCommentController = new PrivateCommentController(datastore);
+
+    // get user
+    UserData user = userController.getUserByID(userID);
+
+    // get this user's tasks
+    ArrayList<TaskData> tasks = taskController.getTasksByUserID(userID);
+
+    // init array of color task will appear
+    String[] colors = new String[tasks.size()];
+
+    // build list of private comments and colors
+    ArrayList<PrivateCommentData> privateComments = new ArrayList<>();
+
+    // only fill if the user is viewing their own profile
+    if (userID == currentUser) {
+      for (int i = 0; i < tasks.size(); i++) {
+        Long taskID = tasks.get(i).getTaskID();
+        PrivateCommentData privateComment =
+            privateCommentController.getPrivateCommentByTaskID(taskID);
+        privateComments.add(privateComment);
+        String statusString = tasks.get(i).getStatus().name();
+        if (statusString.equals("COMPLETE")) {
+          colors[i] = "green";
+        } else {
+          colors[i] = "red";
+        }
+      }
+    }
+
+    // Set attributes of request; retrieve in jsp with
+    // request.setAttribute("userTasks", tasks); I don't need since I can pass from /user-profile
+    request.setAttribute("colors", colors);
+    request.setAttribute("privateComments", privateComments);
+
+    // Load jsp for user page
+    request.getRequestDispatcher("user-private-comments.jsp").forward(request, response);
```

Comment:
Are these going to a new page?  I thought these were going to be displayed in the User Profile "task checklist"?

Reply:
I'm going to do what Tzirliya did with list-tasks.jsp and call that in user_profile.jsp